Add unpack goal (#47).

dranuhl opened the pull request:
Adding rpm:unpack goal.

Features:
- preserve POSIX file permissions if supported by FS, otherwise just read/write
- preserve last modification time if supported by OS
- optional preserve user (POSIX and/or ACL) and group ownership (POSIX only) if supported by FS
- checks file names and prevents path traversal and bad characters to be used

Missing:
- does not support symbolic links, but writes details to debug log about them. Could be added for supporting FS (most *nix and NTFS)
- only directories and regular files are supported currently, partially due to Java Nio limitations
- does not support ACL file-system permissions
- unit and/or integration tests

Note:
- only tested on non-POSIX file system so far
- did not had RPM with symbolic links (or other non-regular files) available, so was not able to test this

ctron commented:
This is awesome! Thanks!

I think it would be great if you could add the known limitations to the documentation too. So that people understand what they can expect.

Also please add license header to the file you added.

dranuhl commented:
- added EPL-2.0 license header.
- added limitation / restrictions to class JavaDoc.
- added file name validation details to class JavaDoc.
- added public setter methods to avoid need to override accessibility.

ctron reviewed (approved):
Looks good to me. Are you ok merging this?

dranuhl commented:
Sure.

ctron commented:
@dranuhl do you want me to create a new release? Or does it make sense to wait for another PR? :heart:

dranuhl commented:
@ctron, have just created a new PR #58  for issue #56.  So unless anyone objects, I guess it would be fine to just create a release after that has been reviewed and merged as well.
